src/Microsoft.Diagnostics.ExtensionCommands/EEHeapCommand.cs

```diff
@@ -0,0 +1,517 @@
+﻿using Microsoft.Diagnostics.DebugServices;
+using Microsoft.Diagnostics.Runtime;
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Diagnostics.ExtensionCommands
+{
+    [Command(Name = "eeheap", Help = "Displays information about native memory that CLR has allocated.")]
+    public class EEHeapCommand : CommandBase
+    {
+        [ServiceImport]
+        public IRuntimeService RuntimeService { get; set; }
+
+        [ServiceImport]
+        public IMemoryService MemoryService { get; set; }
+
+        [Option(Name = "--gc", Aliases = new string[] { "-gc" }, Help = "Only display the GC.")]
+        public bool ShowGC { get; set; }
+
+        [Option(Name = "--loader", Aliases = new string[] { "-loader" }, Help = "Only display the Loader.")]
+        public bool ShowLoader { get; set; }
+
+        public override void Invoke()
+        {
+            IRuntime[] runtimes = RuntimeService.EnumerateRuntimes().ToArray();
```

Comment:
I'm wondering if you really want to total all the runtimes in the system. The main scenario that there is more than one is a .NET Core and a desktop Framework in some web apps.  Most of the other commands use the "current" runtime (set with the `runtimes` command).  I don't have a good sense either way but I thought I bring it up.  If you did want the current runtime just add a ServiceImport property of the type IRuntime.

Reply:
AppFabric has both runtimes loaded for normal operation of their product too, but it is an overall rare scenario to have more than one.

I'm actually of the opinion that most commands should work on all CLR runtimes.  If I'm debugging an application with two CLR, I actually want to know everything. It's hard to imagine if I'm looking for all uses of HttpClient, for example, that I wouldn't want `!dumpheap -type HttpClient` to give me the uses in all runtimes in the process...  If I'm looking at the loader heaps/gc heaps, I probably want to know where _all_ segments are...not just one runtime.

I could be convinced this is a bad idea though.

Reply:
I guess to me if there are two runtimes, the GC commands should be for the current runtime because they are separate GCs with separate stats.  The other reason is to be compatible with the existing commands like !dumpheap and !gcroot.  Those could be changed to enumerate all the runtime but that is a lot harder because they are still implemented in the native C++ code. 

We could have an "--all" option to enumerate all of the runtimes and default to the current one.  In the end, @Maoni0 and/or @cshung probably have a better idea of the best path.

Reply:
it would be useful to enumerate all runtimes. considering the fact that's not so common, if I needed to rank the priority I would rank that support as lower priority than providing ways for the user to filter the output from the current runtime like giving it heap number.

Reply:
Ok. I'm fine with leaving this (your new eeheap command) alone and enumerating all the runtimes.  And I'll start thinking about how to do the same to the native dumpheap/gcroot commands.

Reply:
I think consistently having a `-runtime [desktop|core]` and `-gcheap #` across many commands will help.  I'm taking a look at re-implementing !verifyheap, !gcwhere, !listnearobj, !dumpheap, and similar commands in C# on ClrMD this month.  (I can't promise I'll get to all of them in that time.)  As I do, I'll make sure to add a filter command to keep them consistent.

We can chat about it in the next PR, I'll probably do either !lno or !verifyheap next.